modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/CommandProcessor.java

```diff
@@ -1075,9 +1077,15 @@ private static QueryEntity toQueryEntity(GridSqlCreateTable createTbl) {
 
             res.setKeyFieldName(pkCol.columnName());
         }
-        else
+        else {
             res.setKeyFields(createTbl.primaryKeyColumns());
 
+            if (IgniteFeatures.allNodesSupports(ctx, F.view(ctx.discovery().allNodes(),
+                IgniteDiscoverySpi.SRV_NODES), IgniteFeatures.SPECIFIED_SEQ_PK_KEYS)
+            )
```

Comment:
Should we check if we are in the middle of rolling upgrade?

Reply:
IgniteFeatures.allNodesSupports already handles this case